templates/FFV_form_factors.cpp.in

```diff
@@ -239,12 +365,214 @@ double OneLoopFunctionF(double r)
               0.1875000000000000000  * y * y * y * y * y * y * y * y * y * y * y * y * y +
               0.1764705882352941176  * y * y * y * y * y * y * y * y * y * y * y * y * y * y -
               0.16666666666666666667 * y * y * y * y * y * y * y * y * y * y * y * y * y * y * y);
+   } else if (1./r < ratiolimit) {
+      // Expand about 0 to 1st order
+      return 1.5*r;
    } else {
       return 3. / 2. * (-3.0 + 4.0 * r - r * r - 2.0 * std::log(r)) /
-             pow(1.0 - r, 3);
+             Power3(1.0 - r);
+   }
+}
+
+double OneLoopFunctionH(double r)
+{
+   if (is_zero(r)) {
+      return 5.0;
+   } else if (is_zero(1.0 - r)){
+      return 0.0;
+   } else if (1./r < Power3(ratiolimit)) {
+      // Loop function is about 10^-13, treat as reaching the 0 limit
+      return 0.;
+   } else if (1./r < ratiolimit) {
+      // Expand about 0 to 1st order
+      return r*(-5.-6.*std::log(r));
+   } else {
+      return (5.0 - 27.0 * r + 27.0 * r * r - 5.0 * r * r * r + 6.0 * (r - 3.0) * r * r * std::log(r)) /
+              Power4(1.0 - r);
+   }
+}
+
+double OneLoopFunctionI(double r)
+{
+   if (is_zero(r)) {
+      return 1.0;
+   } else if (is_zero(1.0 - r)){
+      return 0.5;
+   } else if (1./r < Power3(ratiolimit)) {
+      // Loop function is about 10^-13, treat as reaching the 0 limit
+      return 0.;
+   } else if (1./r < ratiolimit) {
+      // Expand about 0 to 1st order
+      return r*(-1.-1.*std::log(r));
+   } else {
+      return (1.0 - r + r * std::log(r)) / Power2(1.0 - r);
+   }
+}
+
+double OneLoopFunctionJ(double r)
```

Comment:
Please add a reference for this function.

Reply:
As above.